plugins/module_utils/rds.py

```diff
@@ -233,3 +238,46 @@ def ensure_tags(client, module, resource_arn, existing_tags, tags, purge_tags):
             parameters={'ResourceName': resource_arn, 'TagKeys': tags_to_remove}
         )
     return changed
+
+
+def compare_iam_roles(existing_roles, target_roles, purge_roles):
```

Comment:
We need some unit testing for these new options.

Reply:
I have started adding some units in this PR https://github.com/ansible-collections/amazon.aws/pull/553

Reply:
I figured.. how can I go about adding unit tests without causing a conflict with your PR? I guess I could copy the parts that I need but if you change anything it'll conflict..

Reply:
@alinabuzachis @jillr I added some unit testing using https://github.com/ansible-collections/amazon.aws/blob/main/tests/unit/module_utils/test_tagging.py as a ref since the methods are similar. I only added for `compare_iam_roles`. I couldn't figure out a good way to unit test `ensure_iam_roles`, but I figured it makes more sense for the integration tests to cover that method instead